### PR TITLE
Fix #1970 appropriate text displayed when there is no result

### LINKED ISSUE #1970: Empty screen is displayed when there are no results in the search
**Actual Behaviour**

Currently when we search for something that is not there in the list, an empty screen is displayed
in various fragments that have a searchView for eg Speakers fragment,Tracks Fragment,schedule fragment,Location fragment

**Expected Behaviour**

A textView should display that there are no results when the user searches for something that is not there in the list
**Steps to reproduce it**

Open the app.
select speakers fragment from the navigation drawer
Write something random in the search view that is not there in the list

**Screenshots of the issue**

screenshot_20171031-211048

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by nikit19**
Fixes #1970 
Changes: appropriate text is displayed when there is no result in tracks fragment,speakers fragment,location fragment ,schedule fragment ,TrackSessions Activity and Location Activity

Screenshots for the change: 
ezgif com-video-to-gif 15

### CODE REVIEW

**Comment by scandindian** [comment]
@nikit19 Looks nice. Maybe we could display text in the center and a bit bigger.

**Comment by nikit19** [comment]
@heysadboy done with the requested changes
screenshot_20171101-221443

File: android/app/src/main/java/org/fossasia/openevent/fragments/LocationsFragment.java [hunk 1]
```diff
@@ -118,6 +120,17 @@ private void handleVisibility() {
             locationsRecyclerView.setVisibility(View.GONE);
         }
     }
+    private void noResult() {
+        if (locationsListAdapter.getItemCount()!=0) {
+            noResultsView.setVisibility(View.GONE);
+            locationsRecyclerView.setVisibility(View.VISIBLE);
+        } else {
+            if (noMicrolocationsView.getVisibility() != View.VISIBLE) {
```
Comment (scandindian) [inline android/app/src/main/java/org/fossasia/openevent/fragments/LocationsFragment.java]: Why are you putting it as else { if ( ) -- . Maybe you could write it as else if () { -- }

File: android/app/src/main/java/org/fossasia/openevent/fragments/SpeakersListFragment.java [hunk 2]
```diff
@@ -132,6 +134,18 @@ private void handleVisibility() {
         }
     }
 
+    private void noResult() {
+        if (speakersListAdapter.getItemCount()!=0) {
+            noSpeakersResultView.setVisibility(View.GONE);
+            speakersRecyclerView.setVisibility(View.VISIBLE);
+        } else {
+            if (noSpeakersView.getVisibility() != View.VISIBLE) {
+                noSpeakersResultView.setVisibility(View.VISIBLE);
```
Comment (scandindian) [inline android/app/src/main/java/org/fossasia/openevent/fragments/SpeakersListFragment.java]: Change nesting statement at other places and here too.

**Review by scandindian — CHANGES_REQUESTED** [review]
Please see the inline comments and update if else statements.

**Comment by nikit19** [comment]
@heysadboy done with the requested changes. Please review

**Review by scandindian — APPROVED** [review]

**Comment by scandindian** [comment]
@nikit19 Fix failing checks too please.

**Comment by nikit19** [comment]
@heysadboy all failing checks fixed

**Comment by scandindian** [comment]
@the-dagger @niranjan94 Please see.

File: android/app/src/main/java/org/fossasia/openevent/fragments/LocationsFragment.java [hunk 3]
```diff
@@ -118,6 +120,15 @@ private void handleVisibility() {
             locationsRecyclerView.setVisibility(View.GONE);
         }
     }
+    private void noResult() {
```
Comment (the-dagger) [inline android/app/src/main/java/org/fossasia/openevent/fragments/LocationsFragment.java]: Can we make this a util function with the views and values as params?

**Review by the-dagger — CHANGES_REQUESTED** [review]

**Comment by nikit19** [comment]
@the-dagger done with the requested changes. Please review

**Review by the-dagger — APPROVED** [review]

**Pull request merged**